lab-kyle/lib/storage.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+// const mkdirp = require('mkdirp');
+const del = require('del');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item){
+  if (!schemaName) return Promise.reject(new Error('expected schema'));
+  if (!item) return Promise.reject(new Error('expected item'));
+  if (!item.name && !item.color) return Promise.reject(new Error('item is not valid'));
+
+  let json = JSON.stringify(item);
+  // return mkdirp(`${__dirname}/../data/${schemaName}/`, function() {
+  return fs.writeFileProm(`${__dirname}/../data/${schemaName}/${item.id}.json`, json)
+    .then( () => item)
+    .catch( err => Promise.reject(err));
+  // });
+};
+
+
+exports.fetchItem = function(schemaName, id) {
+  if (!schemaName) return Promise.reject(new Error('expected schema'));
+  if (!id) return Promise.reject(new Error('expected id'));
+
+  return fs.readFileProm(`${__dirname}/../data/${schemaName}/${id}.json`)
+    .then( data => {
+      try {
+        let item = JSON.parse(data.toString());
+        return item;
+      } catch (err){
+        return Promise.reject(err);
+      }
+    })
+    .catch( err => Promise.reject(err));
+};
+
+exports.fetchAll = function(schemaName) {
+  return fs.readdirProm(`${__dirname}/../data/${schemaName}`)
```

Comment:
Design decision, but I'd like to have all of the contents returned from the files, rather than just the file names.